### PR TITLE
Upgrades to Report UI

### PR DESCRIPTION
**Pull request by Advayp**
Author: Advay

## What changes were made?

See title. Note: the functionality regarding closing and modifying issues has not been implemented, this PR only improves the UI of the report dashboard.

### CODE REVIEW

File: src/components/admin/ReportView.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                  </Link>
+                </ChakraLink>
+              </Text>
+              <ReportStatusView status={status} />{' '}
+              <ReportTypeView type={type} />
+            </HStack>
+            <Divider mt={2} />
+          </ModalHeader>
+          <ModalCloseButton></ModalCloseButton>
+          <ModalBody fontSize={'medium'}>
```
Comment (elimelt) [inline src/components/admin/ReportView.tsx:74]: fontSize="medium"

File: src/components/admin/ReportView.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            name={resolveName(reporter)}
+            src={reporter.profilePictureUrl}
+          />
+        </HStack>
+      </Box>
+      <Modal size={'lg'} isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>
+            <HStack fontSize={'large'}>
```
Comment (elimelt) [inline src/components/admin/ReportView.tsx:59]: same

File: src/components/admin/ReportView.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            <ReportTypeView type={type} />
+          </HStack>
+          <Avatar
+            alignSelf={'flex-end'}
+            name={resolveName(reporter)}
+            src={reporter.profilePictureUrl}
+          />
+        </HStack>
+      </Box>
+      <Modal size={'lg'} isOpen={isOpen} onClose={onClose}>
```
Comment (elimelt) [inline src/components/admin/ReportView.tsx:55]: same

File: src/components/admin/ReportView.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        onClick={onOpen}
+      >
+        <HStack justifyContent={'space-between'}>
+          <HStack>
+            <Text fontWeight={'semibold'}>{reason}</Text>
+            <ReportStatusView status={status} />
+            <ReportTypeView type={type} />
+          </HStack>
+          <Avatar
+            alignSelf={'flex-end'}
```
Comment (elimelt) [inline src/components/admin/ReportView.tsx:49]: same

File: src/components/admin/ReportView.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        borderWidth={1}
+        borderRadius="lg"
+        p={4}
+        boxShadow="md"
+        cursor={'pointer'}
+        onClick={onOpen}
+      >
+        <HStack justifyContent={'space-between'}>
+          <HStack>
+            <Text fontWeight={'semibold'}>{reason}</Text>
```
Comment (elimelt) [inline src/components/admin/ReportView.tsx:44]: same

File: src/components/admin/ReportView.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      <Box
+        key={key}
+        borderWidth={1}
+        borderRadius="lg"
+        p={4}
+        boxShadow="md"
+        cursor={'pointer'}
+        onClick={onOpen}
+      >
+        <HStack justifyContent={'space-between'}>
```
Comment (elimelt) [inline src/components/admin/ReportView.tsx:42]: same

File: src/components/admin/ReportView.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+  return (
+    <>
+      <Box
+        key={key}
+        borderWidth={1}
+        borderRadius="lg"
+        p={4}
+        boxShadow="md"
+        cursor={'pointer'}
```
Comment (elimelt) [inline src/components/admin/ReportView.tsx:39]: same

File: src/components/admin/ReportTypeView.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        return 'pink';
+      case ReportType.Member:
+        return 'yellow';
+      case ReportType.Question:
+        return 'cyan';
+    }
+  };
+
+  return (
+    <Badge borderRadius={'md'} colorScheme={getColorScheme(type)}>
```
Comment (elimelt) [inline src/components/admin/ReportTypeView.tsx:22]: can be borderRadius="md"

File: src/components/admin/ReportStatusView.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        return 'green';
+      case ReportStatus.Pending:
+        return 'purple';
+      case ReportStatus.Resolving:
+        return 'blue';
+    }
+  };
+
+  return (
+    <Badge borderRadius={'md'} colorScheme={getColorScheme(status)}>
```
Comment (elimelt) [inline src/components/admin/ReportStatusView.tsx:22]: borderRadius="md"

File: src/components/admin/ReportedInterview.tsx [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+  interview: HydratedInterview;
+}
+
+export const ReportedInterview: React.FC<Props> = ({ interview }) => {
+  const technicalQuestions = interview.technicalQuestions ?? [];
+  const behavioralQuestions = interview.behavioralQuestions ?? [];
+
+  return (
+    <>
+      <Text fontWeight={'semibold'}>
```
Comment (elimelt) [inline src/components/admin/ReportedInterview.tsx:17]: fontWeight="semibold"

File: src/components/admin/ReportedInterview.tsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+        </Link>
+      </Text>
+      <Text fontWeight={'semibold'}>
+        Interviewee:{' '}
+        <Link to={`/directory/${interview.interviewee.id}`}>
+          @{interview.interviewee.username}
+        </Link>
+      </Text>
+      {technicalQuestions.length > 0 && (
+        <Text fontWeight={'semibold'}>
```
Comment (elimelt) [inline src/components/admin/ReportedInterview.tsx:30]: fontWeight="semibold"

File: src/components/admin/ReportedInterview.tsx [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+  return (
+    <>
+      <Text fontWeight={'semibold'}>
+        Interviewer:{' '}
+        <Link to={`/directory/${interview.interviewer.id}`}>
+          @{interview.interviewer.username}
+        </Link>
+      </Text>
+      <Text fontWeight={'semibold'}>
```
Comment (elimelt) [inline src/components/admin/ReportedInterview.tsx:23]: fontWeight="semibold"

File: src/components/admin/ReportedInterview.tsx [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+              textAlign={'center'}
+              key={idx}
+            >
+              {question.title} <ExternalLinkIcon fontSize={'small'} />
+            </ChakraLink>
+          ))}
+        </Text>
+      )}
+      {behavioralQuestions.length > 0 && (
+        <Text fontWeight={'semibold'}>
```
Comment (elimelt) [inline src/components/admin/ReportedInterview.tsx:47]: fontWeight="semibold"

File: src/components/admin/ReportedInterview.tsx [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+          Behavioral Questions:{' '}
+          {behavioralQuestions.map((question, idx) => (
+            <Text fontWeight={'normal'} key={idx}>
```
Comment (elimelt) [inline src/components/admin/ReportedInterview.tsx:50]: fontWeight="normal"

File: src/components/admin/ReportedMember.tsx [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+import { Text } from '@chakra-ui/react';
+import { Link } from 'react-router-dom';
+
+interface Props {
+  member: Member;
+}
+
+export const ReportedMember: React.FC<Props> = ({ member }) => {
+  return (
+    <Text fontWeight={'semibold'}>
```
Comment (elimelt) [inline src/components/admin/ReportedMember.tsx:12]: fontWeight="semibold"

File: src/components/admin/ReportedQuestion.tsx [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+import { formatDate } from '../../localization';
+
+interface Props {
+  question: TechnicalQuestion;
+}
+
+export const ReportedQuestion: React.FC<Props> = ({ question }) => {
+  return (
+    <>
+      <Text fontWeight={'semibold'}>
```
Comment (elimelt) [inline src/components/admin/ReportedQuestion.tsx:14]: fontWeight="semibold"

File: src/components/admin/ReportedQuestion.tsx [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+        Reported question:{' '}
+        <ChakraLink
+          href={question.source}
+          fontWeight="normal"
+          color={'blue.400'}
```
Comment (elimelt) [inline src/components/admin/ReportedQuestion.tsx:19]: x="y"

File: src/components/admin/ReportedQuestion.tsx [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+          textAlign={'center'}
```
Comment (elimelt) [inline src/components/admin/ReportedQuestion.tsx:20]: x="y"

File: src/components/admin/ReportedQuestion.tsx [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+        >
+          {question.title} <ExternalLinkIcon fontSize={'small'} />
```
Comment (elimelt) [inline src/components/admin/ReportedQuestion.tsx:22]: x="y"

File: src/components/admin/ReportedQuestion.tsx [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+        </ChakraLink>
+      </Text>
+      <HStack gap={1} alignItems={'flex-start'}>
```
Comment (elimelt) [inline src/components/admin/ReportedQuestion.tsx:25]: x="y"

File: src/components/admin/ReportedQuestion.tsx [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+        <Text fontWeight="semibold">Approved by:</Text>
+        <Text>
+          {question.approvedBy ? `@${question.approvedBy}` : 'No one :('}
+        </Text>
+      </HStack>
+      <HStack gap={1} alignItems={'flex-start'}>
```
Comment (elimelt) [inline src/components/admin/ReportedQuestion.tsx:31]: x="y"

**Review by elimelt — COMMENTED** [review]
me when I get paid per comment

**Comment by Advayp** [comment]
> me when I get paid per comment

Lol

**Review by elimelt — APPROVED** [review]

**Pull request merged**